tests/UploadControllerTest.cs

```diff
@@ -144,6 +144,38 @@ public async Task UploadShouldSaveMinimalDatasetAsync()
         Assert.AreEqual(null, workflow.Finished);
     }
 
+    [TestMethod]
+    public async Task UploadShouldSaveBoreholeWithAttachmentsAsync()
+    {
+        httpClientFactoryMock
+           .Setup(cf => cf.CreateClient(It.IsAny<string>()))
+           .Returns(() => new HttpClient())
+           .Verifiable();
+
+        var csvFile = "borehole_with_attachments.csv";
+
+        byte[] fileBytes = File.ReadAllBytes(csvFile);
+        using var stream = new MemoryStream(fileBytes);
+
+        var boreholeCsvFormFile = new FormFile(stream, 0, fileBytes.Length, null, csvFile);
+
+        var firstPdfFile = "borehole_attachment_1.pdf";
+        fileBytes = File.ReadAllBytes(firstPdfFile);
+        var firstPdfFormFile = new FormFile(new MemoryStream(File.ReadAllBytes(firstPdfFile)), 0, fileBytes.Length, null, "borehole_attachment_1.pdf");
+
+        var secondPdfFile = "borehole_attachment_2.pdf";
+        fileBytes = File.ReadAllBytes(secondPdfFile);
+        var secondPdfFormFile = new FormFile(new MemoryStream(File.ReadAllBytes(secondPdfFile)), 0, fileBytes.Length, null, "borehole_attachment_2.pdf");
+
+        ActionResult<int> response = await controller.UploadFileAsync(workgroupId: 1,
+                                                                     boreholesFile: boreholeCsvFormFile,
+                                                                     pdfAttachments: new List<IFormFile>() { firstPdfFormFile, secondPdfFormFile });
```

Comment:
PP: Ich würde nur die `workgroupId` benennen und den Rest nicht, da sie selbstredend sind. Beispiel:
```suggestion
        ActionResult<int> response = await controller.UploadFileAsync(workgroupId: 1, boreholeCsvFormFile, new List<IFormFile> { firstPdfFormFile, secondPdfFormFile });
```